code/datums/singularities.dm

```diff
@@ -0,0 +1,310 @@
+/*
+Singularity behaviors, for all your singularity needs.
+
+This file contains the base type and:
+- Normal Singularity
++ Katamari (normal singulo sub-type)
+*/
+
+///
+
+#define EVENT_GROWTH 3//the rate at which the event proc radius is scaled relative to the radius of the singularity
+#define EVENT_MINIMUM 5//the base value added to the event proc radius, serves as the radius of a 1x1
+
+ABSTRACT_TYPE(/datum/singularity_behavior/)
+/datum/singularity_behavior/
+	/// What singularity object is using me?
+	var/obj/machinery/the_singularity/singularity
+	/// Use the warpy effect from modern singularity. Props to amylizzle!
+	var/use_amylizzle_animation = TRUE
```

Comment:
I feel like a var name like this isn't particularly descriptive. Knowing who made the animation doesn't help the person reading the code. And also it is not animated so calling it an animation is a misnomer. I'd probably name this var something like `use_distortion_effect`